bin/build_llvm-legacy.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+#
+#  build_llvm-legacy.sh:  Script to build the legacy LLVM used by flang-legacy. binary driver
+#         This driver will never call flang -fc1, it only calls binaries
+#             clang, flang1, flang2, built elsewhere
+#  Instead of downloading the ROCm 5.5 llvm package we have to
+#  compile the 11vm/clang libs from source to support various
+#  operating systems and spack. This will be the llvm-legacy build step.
+#  These libs/headers are not installed and will picked up from the build
+#  tree for flang-legacy.
+#
+BUILD_TYPE=${BUILD_TYPE:-Release}
+
+# --- Start standard header to set AOMP environment variables ----
+realpath=`realpath $0`
+thisdir=`dirname $realpath`
+. $thisdir/aomp_common_vars
+# --- end standard header ----
+
+if [ $AOMP_BUILD_FLANG_LEGACY == 0 ] ; then
+   if [ "$1" != "install" ] ; then
+      echo "WARNING:  ROCM install for $AOMP_FLANG_LEGACY_REL/llvm-legacy not found."
+      echo "          This build will skip build of flang-legacy."
+      echo "          The flang will link to the clang driver."
+   fi
+   exit
+fi
+TARGETS_TO_BUILD="AMDGPU;${AOMP_NVPTX_TARGET}X86"
+
+if [ $AOMP_STANDALONE_BUILD == 1 ] ; then
+   standalone_word="_STANDALONE"
+else
+   standalone_word=""
+fi
+
+if [ "$AOMP_USE_NINJA" == 0 ] ; then
+    AOMP_SET_NINJA_GEN=""
+else
+    AOMP_SET_NINJA_GEN="-G Ninja"
+fi
+osversion=$(cat /etc/os-release | grep -e ^VERSION_ID)
+if [[ $osversion =~ '"7.' ]] || [[ $osversion =~ '"8' ]]; then
+  _cxx_flag="-DCMAKE_CXX_FLAGS='-D_GLIBCXX_USE_CXX11_ABI=0'"
+else
+  _cxx_flag=""
+fi
+
+# Legacy Flang dosen't support building of compiler-rt so it
+# utilizes the clang runtime libraries build/install using build_project.sh.
+# The LLVM_VERSION_MAJOR of legacy flang driver has to match with the clang
+# binaries generated from build_project.sh.
+LLVM_VERSION_MAJOR=$(${AOMP}/bin/clang --version | grep -oP '(?<=clang version )[0-9]+')
```

Comment:
After rebasing your branch on top of recent `aomp-dev` to get the recent build script updates, I noticed one change is needed to fix a build issue in `llvm-legacy`. `${AOMP}/bin/clang` needs updated to `${LLVM_INSTALL_LOC}/bin/clang`. Other than that AOMP built fine.

Reply:
Thank you! I'll push a new version with that fix.